src/lib/dns.coffee

```diff
@@ -143,7 +143,9 @@ module.exports = (dnschain) ->
             if (resolver = @dnschain.chainsTLDs[q.name.split('.').pop()])
                 @log.debug gLineInfo("resolving via #{resolver.name}..."), {domain:q.name, q:q}
 
-                @dnschain.cache.resolveBlockchain resolver, q.name, {}, (err, result) =>
+                return @sendErr(res, NAME_RCODE.SERVFAIL, cb) if not resolver.resources.key?
```

Comment:
Maybe add a `@log.warn` message to this `if` block so that the sysadmin knows why the operation failed?